plugins/woocommerce/includes/admin/class-wc-admin-menus.php

```diff
@@ -284,22 +329,41 @@ public function menu_order( $menu_order ) {
 		// Initialize our custom order array.
 		$woocommerce_menu_order = array();
 
+		// Cache HPOS status check.
+		$is_hpos_enabled = wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled();
+
 		// Get the index of our custom separator.
 		$woocommerce_separator = array_search( 'separator-woocommerce', $menu_order, true );
 
 		// Get index of product menu.
 		$woocommerce_product = array_search( 'edit.php?post_type=product', $menu_order, true );
 
+		// Get index of orders menu.
+		$woocommerce_orders = array_search( 'wc-orders', $menu_order, true );
+
+		// Also check for legacy post type orders menu when HPOS is disabled.
+		if ( false === $woocommerce_orders && ! $is_hpos_enabled ) {
+			$woocommerce_orders = array_search( 'edit.php?post_type=shop_order', $menu_order, true );
+		}
```

Comment:
Okay, so HPOS check is needed to differentiate between post_type orders in the menu and non post_type orders... 🤔. What's not clear (and I haven't dug deeply) is why these conditional checks are needed now, but weren't for the current menu implementation?

Reply:
Here specifically it's because the "Orders" item was never sorted, but the menu item has a different slug depending on if it's HPOS or not (custom menu vs post type), so the checks are often different in various places.